video/out/wayland_common.c

```diff
@@ -1552,10 +1564,39 @@ static void apply_keepaspect(struct vo_wayland_state *wl, int *width, int *heigh
     if (!wl->opts->keepaspect)
         return;
 
+    int phys_width = handle_round(wl->scaling, *width);
+    int phys_height = handle_round(wl->scaling, *height);
+
+    // Ensure that the size actually changes before we start trying to actually
+    // calculate anything so the wrong direction isn't choosen.
+    if (wl->resizing && !wl->resizing_direction &&
+        phys_width == mp_rect_w(wl->geometry) && phys_height == mp_rect_h(wl->geometry))
+        return;
+
+    // We are doing a continuous resize (e.g. dragging with mouse), keep the
+    // aspect ratio against the height if the change is only in the height
+    // coordinate.
+    if (wl->resizing && !wl->resizing_direction && phys_width == mp_rect_w(wl->geometry) &&
+        phys_height != mp_rect_h(wl->geometry)) {
+        wl->resizing_direction = MP_HEIGHT_RESIZE;
```

Comment:
Thinking about it more, I think `wl->constrain_dimension = MP_CONSTRAIN_Y` may be clearer

Reply:
I changed it but not exactly to your suggestions. Felt like specifying that this was for `resizing` and against `width/height` was important.

Reply:
Could've done MP_RESIZE_CONSTRAIN_{X,Y} but this works too